### PR TITLE
Capture media config bundle in sync part of createOffer/Answer

### PR DESCRIPTION
**Pull request by adam-be**
Proposed fix for #782 

Perhaps it would be useful to include the example in the issue in the spec to highlight this as well.

### CODE REVIEW

File: webrtc.html [hunk 1]
```diff
@@ -1293,6 +1293,27 @@
...
+                    information MUST be captured:</p>
+                    <ul>
+                      <li>
+                        <p>If <code><a href=
+                        "#dom-rtcpeerconnection-createdatachannel">createDataChannel()</a></code>
+                        has been called for the first time.</p>
+                      </li>
+                      <li>
+                        <p>All <code><a>RTCRtpSender</a></code> objects that
+                        have been added to or removed from
```
Comment (pthatcherg) [inline webrtc.html]: This should be in terms of RtpTransceivers, not RtpSenders.

File: webrtc.html [hunk 2]
```diff
@@ -1293,6 +1293,27 @@
...
+                    collection of information that captures the changes made to
+                    the media configuration of <var>connection</var> since the
+                    last time an offer or answer was created, or since the
+                    creation of <var>connection</var> if no offer or answer has
+                    been created prior to this call. The following pieces of
+                    information MUST be captured:</p>
+                    <ul>
+                      <li>
+                        <p>If <code><a href=
+                        "#dom-rtcpeerconnection-createdatachannel">createDataChannel()</a></code>
```
Comment (pthatcherg) [inline webrtc.html]: Perhaps this should be in terms of SctpTransport, which is like the data channel version of RtpTransceiver.

File: webrtc.html [hunk 3]
```diff
@@ -1293,6 +1293,27 @@
                     method was invoked.</p>
                   </li>
                   <li>
+                    <p>Let <var>offerBundle</var> be an implementation specific
+                    collection of information that captures the changes made to
+                    the media configuration of <var>connection</var> since the
+                    last time an offer or answer was created, or since the
+                    creation of <var>connection</var> if no offer or answer has
+                    been created prior to this call. The following pieces of
+                    information MUST be captured:</p>
```
Comment (pthatcherg) [inline webrtc.html]: This seems like a complex way of expressing this.

File: webrtc.html [hunk 4]
```diff
@@ -1450,7 +1493,8 @@
                       </li>
                       <li>
                         <p>In parallel, start the process to generate an SDP
-                        answer, as described in <span data-jsep=
+                        answer that reflects changes in <var>answerBundle</var>,
+                        as described in <span data-jsep=
                         "generating-an-answer">[[!JSEP]]</span>.</p>
```
Comment (pthatcherg) [inline webrtc.html]: This sounds like we have to rewrite JSEP in a major way to make this work.

Comment (pthatcherg) [inline webrtc.html]: What happens with this?

setRemoteDescription(offer);
createAnswer();

Does that work?  

setRemoteDescription(offer);
createAnswer();
addTrack();

Does it affect setRemoteDescription?  Does it affect createAnswer?


I think this isn't going to work.

**Comment by alvestrand** [comment]
Based on discussions at TPAC, this is not the way forward. Closing.

**Comment by alvestrand** [comment]
#820 is the new attempt.

**Comment by adam-be** [comment]
This PR was replaced by by #820.

**Pull request closed without merging**